test/headers-timeout.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+const { test } = require('tap')
+const { Client, errors } = require('..')
+const { createServer } = require('net')
+
+const nodeMajorVersion = parseInt(process.version.split('.')[0].slice(1))
+
+if (nodeMajorVersion >= 14) {
+  test('headers timeout', (t) => {
+    t.plan(1)
+
+    const server = createServer((socket) => {
```

Comment:
I don't see much difference here from what is implemented  in the Node tests you posted